modules/stochastic_tools/src/userobject/PolynomialChaos.C

```diff
@@ -186,49 +142,38 @@ PolynomialChaos::computeSTD() const
     Real norm = 1.0;
     for (std::size_t d = 0; d < _ndim; ++d)
       norm *= _poly[d]->innerProduct(_tuple[i][d]);
-    var += pow(_coeff[i], 2) * norm;
+    var += _coeff[i] * _coeff[i] * norm;
   }
 
   return std::sqrt(var);
 }
 
 Real
-PolynomialChaos::computeSkewness() const
-{
-  Real mu = computeMean();
-  Real sig = computeSTD();
-  return (powerExpectation(3) - 3.0 * Utility::pow<2>(sig) * mu - Utility::pow<3>(mu)) /
-         Utility::pow<3>(sig);
-}
-
-Real
-PolynomialChaos::computeKurtosis() const
-{
-  Real mu = computeMean();
-  Real sig = computeSTD();
-  Real e3 = powerExpectation(3);
-  Real e4 = powerExpectation(4);
-  return (e4 - 4 * e3 * mu + 6.0 * Utility::pow<2>(sig) * Utility::pow<2>(mu) +
-          3.0 * Utility::pow<4>(mu)) /
-         Utility::pow<4>(sig);
-}
-
-Real
-PolynomialChaos::powerExpectation(const unsigned int n) const
+PolynomialChaos::powerExpectation(const unsigned int n, const bool distributed) const
 {
   std::vector<StochasticTools::WeightedCartesianProduct<unsigned int, Real>> order;
   for (unsigned int d = 0; d < _ndim; ++d)
   {
     std::vector<std::vector<unsigned int>> order_1d(n);
-    std::vector<std::vector<Real>> c_1d(n, _coeff);
-    for (unsigned int i = 0; i < _ncoeff; ++i)
+    std::vector<std::vector<Real>> c_1d(n);
+    for (unsigned int i = 1; i < _ncoeff; ++i)
       for (unsigned int m = 0; m < n; ++m)
+      {
         order_1d[m].push_back(_tuple[i][d]);
+        c_1d[m].push_back(_coeff[i]);
+      }
     order.push_back(StochasticTools::WeightedCartesianProduct<unsigned int, Real>(order_1d, c_1d));
```

Comment:
Do you know the size of this up front? If so, can you call `order.reserve(num)` after you create it.